xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/viewers/comments.js

```diff
@@ -350,22 +420,91 @@ viewers.Comments = Class.create({
       this.initialLocation.insert({after: this.form});
     }
     this.form["XWiki.XWikiComments_replyto"].value = "";
-    this.form["XWiki.XWikiComments_comment"].value = "";
+    this.reloadEditor();
     this.cancelPreview(this.form);
   },
   /**
    * Registers a listener that watches for the insertion of the Comments tab and triggers the enhancements.
    * After that, the listener removes itself, since it is no longer needed.
    */
-  addTabLoadListener : function(event) {
-    var listener = function(event) {
+  addTabLoadListener: function (event) {
+    var listener = function (event) {
       if (event.memo.id == 'Comments') {
         this.startup();
       }
     }.bindAsEventListener(this);
     document.observe("xwiki:docextra:loaded", listener);
+  },
+  destroyEditor: function (selector, name) {
+
+    // TODO: delete and find a way to directly call the editor destroy event.
+    if (typeof CKEDITOR !== 'undefined' && name in CKEDITOR.instances) {
+      CKEDITOR.instances[name].destroy();
+    }
+
+    require(['jquery'], function ($) {
+      if ($(selector).size() > 0) {
+        const parent = $(selector).parent(".wysiwyg-field");
+        parent.empty();
+        $(document).trigger('xwiki:dom:updated', {'elements': parent.toArray()});
+      }
+    });
+  },
+  reloadEditor : function (options) {
+    var wfClass = '.wysiwyg-field';
+    options = options || {};
+    const commentNbr = options.commentNbr;
+    const callback = options.callback;
+    var name = 'XWiki.XWikiComments_comment';
+    if (commentNbr) {
+      name = 'XWiki.XWikiComments_' + commentNbr + '_comment';
+      wfClass = wfClass + '-' + commentNbr;
+    }
+
+    this.destroyEditor("[name='" + name + "']", name);
+
+    require(['jquery', 'xwiki-events-bridge'], function ($) {
+      if ($(".wysiwyg-field").size() > 0) {
+        $.post(new XWiki.Document().getURL("view") + '?' + $.param({
+          number: commentNbr,
+          xpage: 'xpart',
+          vm: 'commentfield.vm',
+          syntax: 'xwiki/2.1',
```

Comment:
The syntax is always necessarily `xwiki/2.1` we don't authorize to comment in `xwiki/2.0` or `markdown`?